File: GoogleGames/src/RockAndRoll.java
import java.io.File;
import java.io.FileNotFoundException;
import java.util.ArrayList;
import java.util.Scanner;

public class RockAndRoll {

	public static void main(String[] args){
		Scanner sc = new Scanner(System.in);
		//System.out.print("What are the dimensions of the map (h w)? ");
		//int numMapCols = sc.nextInt();
		//int numMapRows = sc.nextInt();
		
		File movementsFile = new File("movements.txt");
		File mapFile = new File("seaMap.txt");
		int[][] map = new int[20][20];
		int[] movements = new int[0];
		try {
			map = getMapFromFile(mapFile, 20, 20);
			movements = getMovementsFromFile(movementsFile);
		} catch (FileNotFoundException e) {
			e.printStackTrace();
		}
		testStartingPosition(map, movements);
	}
	
	public static void testStartingPosition(int[][] map, int[] moves){
		for (int startRow = 0; startRow < map.length; startRow++){
			for (int startCol = 0; startCol < map[0].length; startCol++){
				if (map[startRow][startCol] == 1){
					
				} else {
					int currRow = startRow;
					int currCol = startCol;
					for (int moveIndex = 0; moveIndex < moves.length; moveIndex++){
						int move = moves[moveIndex];
						switch (move){
							case 0: currRow--;
									break;
							case 1: currCol--;
									break;
							case 2: currRow++;
									break;
							case 3: currCol++;
									break;
						}
						if (currRow < 0 || currRow >= map.length || currCol < 0 || currCol >= map[0].length || map[currRow][currCol] == 1){
							System.out.println("Moving to " + currRow + ", " + currCol);
							System.out.println();
							break;
						} else if (moveIndex == moves.length - 1){
						}
					}
				}
			}
		}
	}
	
	public static int[] getMovementsFromFile(File movementsFile) throws FileNotFoundException{
		Scanner num = new Scanner(movementsFile);
		int numMoves = 0;
		while (num.hasNext()){
			num.next();
			numMoves++;
		}
		int[] moves = new int[numMoves];
		Scanner getMoves = new Scanner(movementsFile);
		int index = 0;
		while (getMoves.hasNext()){
			char move = getMoves.next().charAt(0);
			if (move == 'N'){
				moves[index] = 0;
			} else if (move == 'W'){
				moves[index] = 1;
			} else if (move == 'S'){
				moves[index] = 2;
			} else if (move == 'E'){
				moves[index] = 3;
			}
			index++;
		}
		return moves;
	}
	
	public static int[][] getMapFromFile(File mapFile, int numRows, int numCols) throws FileNotFoundException{
		int[][] map = new int[numRows][numCols];
		Scanner mapScanner = new Scanner(mapFile);
		for (int row = 0; row < numRows; row++){
			for (int col = 0; col < numCols; col++){
				if (mapScanner.hasNextInt()){
					map[row][col] = mapScanner.nextInt();
				} else {
					System.out.println("You done fucked up!");
				}
			}
		}
		return map;
	}
}

File: GoogleGames/src/RockAndRoll.py
def create_map(map_name):
    map_file = open(map_name, "r")
    map_list = map_file.read().splitlines()
    row_index = 0
    while row_index < len(map_list):
        row_list = map_list[row_index].split(sep=" ")
        row_list = list(map(int, row_list))
        map_list[row_index] = row_list
        row_index += 1
    print("The following map was constructed from the given file: ")
    for row in map_list:
        for col in row:
            print(col, end=" ")
        print()
    print("With dimensions " + len(map_list).__str__() + " x " + len(map_list[0]).__str__())
    return map_list


def create_movement_list(move_name):
    move_file = open(move_name, "r")
    move_file = move_file.read().split()
    print("The following movements were constructed from the given file: ")
    for move in move_file:
        print(move, end=" ")
    print()
    return move_file


def find_sub_position(map_list, move_list):
    start_row = 0
    while start_row < len(map_list):
        start_col = 0
        while start_col < len(map_list[0]):
            curr_row = start_row
            curr_col = start_col
            move_index = 0
            valid_move = True
            while move_index < len(move_list) and valid_move:
                move = move_list[move_index]
                if move == "N":
                    curr_row -= 1
                elif move == "S":
                    curr_row += 1
                elif move == "E":
                    curr_col += 1
                elif move == "W":
                    curr_col -= 1
                else:
                    print("You done messed up A-aron")
                    return
                if curr_row < 0 or curr_row >= len(map_list) or curr_col < 0 or curr_col >= len(map_list[0]):
                    print([start_row, start_col].__str__() + " failed on " + [curr_row, curr_col].__str__() + " with index out of bounds")
                    valid_move = False
                elif map_list[curr_row][curr_col] == 1:
                    print([start_row, start_col].__str__() + " failed on " + [curr_row, curr_col].__str__() + " with a land position")
                    valid_move = False
                move_index += 1
            if move_index == len(move_list):
                print("Made it all the way through with " + [curr_row, curr_col].__str__())
                return
            start_col += 1
        start_row += 1
    return [-1, -1]


# map_name = input("What file would you like to use as a map? ")
map_arr = create_map("seaMap.txt")

# movements_name = input("What file contains the movements of the enemy sub? ")
move_arr = create_movement_list("movements.txt")
# print(len(move_arr))
sub_coords = find_sub_position(map_arr, move_arr)
print("The following coordinates are for the sub: " + sub_coords.__str__())
